### PR TITLE
Fix 18035

### LINKED ISSUE #18035: 🐛 Bug Report: Profile Picture String /spec/profile/picture must be string
### 📜 Description

We have connected our self-hosted GitLab instance with Backstage and we have setup automatic discovery. It works fine except for users without profile picture.

I go to the Catalog and select Users I will only see users that have a profile picture set.

### 👍 Expected behavior

It should synchronize all users regardless if they have a profile picture set or not

### 👎 Actual Behavior with Screenshots

It actually only synchronizes users with profile pictures and prints a warning for every other user: 

`[1] 2023-05-31T09:25:04.661Z catalog warn Processor BuiltinKindsEntityProcessor threw an error while validating the entity user:default/<name>; caused by TypeError: /spec/profile/picture must be string - type: string type=plugin entity=user:default/<name> location=url:https://<URL>/<name>
`

### 👟 Reproduction steps

1. Setup backstage
2. Define the integration in ```app-config.yaml```
```
catalog:
  providers:
    gitlab:
      yourProviderId:
        host: ${GITLAB_HOST}
        orgEnabled: true
        group: <our-group>
        schedule: # optional; same options as in TaskScheduleDefinition
          # supports cron, ISO duration, "human duration" as used in code
          frequency: { minutes: 2 }
          # supports ISO duration, "human duration" as used in code
          timeout: { minutes: 1 }
```
3. Run ```yarn dev```

### 📃 Provide the context for the Bug.

I bumped to the latest release to make sure no update fixed this already but no help :( 

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by npiyush97**
#### :heavy_check_mark: Checklist

fix #18035 
- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/gorgeous-timers-act.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend-module-gitlab': minor
```
Comment (jhaals) [inline .changeset/gorgeous-timers-act.md:2]:
```suggestion
'@backstage/plugin-catalog-backend-module-gitlab': patch
```

File: .changeset/gorgeous-timers-act.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+added test if user does not have name or profile picture,variable default to undefined
```
Comment (jhaals) [inline .changeset/gorgeous-timers-act.md:5]:
```suggestion
Fixed bug that caused ingestion errors when a user's name or profile picture is undefined.
```

Reworded this slightly as this is what goes in the changelog for this package. The tests are a great addition but an internal thing to the package so no need mentioning that.

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! Some minor tweaks added

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! Some minor tweaks added

**Comment by malagant** [comment]
This is a very important fix. Can you already tell, when this will be published?

**Comment by jamieklassen** [comment]
@malagant according to the github-actions bot post [above](https://github.com/backstage/backstage/pull/18077#issuecomment-1573598035),

> Thank you for contributing to Backstage! The changes in this pull request will be part of the `1.15.0` release, scheduled for Tue, 20 Jun 2023.

it should be published next Tuesday.

**Pull request merged**